composer/utils/object_store/gs_object_store.py

```diff
@@ -0,0 +1,211 @@
+    """
+
+    def __init__(
+        self,
+        gs_root_dir: str,
+    ) -> None:
+        try:
+            from google.cloud.storage import Client
+        except ImportError as e:
+            raise MissingConditionalImportError('gcs', 'google.cloud.storage') from e
+
+        if 'GOOGLE_APPLICATION_CREDENTIALS' in os.environ:
+            service_account_path = os.environ['GOOGLE_APPLICATION_CREDENTIALS']
+            self.client = Client.from_service_account_json(service_account_path)
+        elif 'GCS_KEY' in os.environ and 'GCS_SECRET' in os.environ:
+            # Create a session and use it to make our client. Unlike Resources and Sessions,
+            # clients are generally thread-safe.
+
+            try:
+                import boto3
+            except ImportError as e:
+                raise MissingConditionalImportError('gcs', 'boto3') from e
+
+            session = boto3.session.Session()
+            self.client = session.client('s3',
+                                         region_name='auto',
+                                         endpoint_url='https://storage.googleapis.com',
+                                         aws_access_key_id=os.environ['GCS_KEY'],
+                                         aws_secret_access_key=os.environ['GCS_SECRET'])
+        else:
+            raise ValueError(f'GOOGLE_APPLICATION_CREDENTIALS needs to be set for ' +
+                             f'service level accounts or GCS_KEY and GCS_SECRET env variables must be set.')
+
+        from composer.utils import parse_uri
+        backend, self.bucket_name, self.prefix = parse_uri(gs_root_dir)
+        if backend == '':
+            raise ValueError(f"gs_root_dir ({gs_root_dir}) doesn't have a valid format")
+        self.prefix = self.prefix.lstrip('/')
+
+        try:
+            self.bucket = self.client.get_bucket(self.bucket_name, timeout=60.0)
+        except Exception as e:
+            _reraise_gs_errors(gs_root_dir, e)
+
+    def get_key(self, object_name: str) -> str:
+        return f'{self.prefix}{object_name}'
+
+    def get_uri(self, object_name: str) -> str:
+        return f'gs://{self.bucket_name}/{self.get_key(object_name)}'
+
+    def get_object_size(self, object_name: str) -> int:
+        """Retrieves the size of an object stored in the cloud storage bucket.
+
+        Args:
+            object_name (str): The name of the object in the cloud storage bucket whose size is to be retrieved.
+
+        Returns:
+            int: The size of the object in bytes.
+
+        Raises:
+            FileNotFoundError: If the specified object does not exist in the cloud storage bucket.
+            Exception: If an error occurs while trying to retrieve the object's size.
+        """
+        from google.cloud.storage import Blob
+
+        key = self.get_key(object_name)
+        stats = Blob(bucket=self.bucket, name=key).exists(self.client)
```

Comment:
more informative variable name please. This is really like `blob_exists` right?

Reply:
what does your comment above (I still think this. Is it not true for some reason?) refer to? Is it moving parse_uri import? It cannot be moved outside this function otherwise there will be a circular import.